File: 프로그래머스/Level 2/짝지어 제거하기.cpp
//https://programmers.co.kr/learn/courses/30/lessons/12973

#include <iostream>
#include<string>
#include <stack>
using namespace std;

int solution(string s)
{
    int answer = 0;
    stack<char> st;
    
    for(char c : s){
        if(st.empty()) st.push(c);
        else{
            if(st.top() == c){
                st.pop();
            }
            else
                st.push(c);
        }
    }
    
    if(st.empty())
        answer = 1;
    // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    cout << "Hello Cpp" << endl;

    return answer;
}

File: 프로그래머스/Level 2/짝지어 제거하기.py
//https://programmers.co.kr/learn/courses/30/lessons/12973?language=python3

def solution(s):
    answer = 0

    st = []
    
    for i in s:
        if not st: #empty
            st.append(i)
        else :  # not empty
            if st[-1] == i:
                st.pop()
            else:
                st.append(i)
                
    if not st:
        answer = 1
    return answer
